### PR TITLE
[#2PlaysAMonth]: Image Gallery - Create a responsive image gallery by using the free Unsplash API

### PR DESCRIPTION
**Pull request by siddhantsiddh15**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

The project contains use of Unsplash API, which is fetched using axios and then populated in the Photo Component.

Following things have been used in this project:

1. useState
2. useEffect
3. React Form
4. Component Refactoring
5. Material UI icon
6. Axios npm module
7. CSS Grid
8. Infinite scroll functionality(This feature is still not showing the react play app but is working on local file system. I am working on it)

The app is responsive and redirects to the individual images and profiles when clicked on it.

Fixes [#910](https://github.com/reactplay/react-play/issues/910)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

To see working of the app repeat the following 
1. Go to the app. Some images are loaded on the screen by default
2. Search for any keyword. The screen will update with new images.
3. Hover over any image, the cursor will turn to pointer.
4. On clicking any image, the page will re direct to that image.
5. On hovering, the profile name, likes will come in front of the image. On clicking the profile photo, the profile of the image contributor will open
6. The page on scrolling down should show more images. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

image

image

image

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
@siddhantsiddh15 , Thanks for the PR.

I would request you change the PR title to "[#2PlaysAMonth]: Image Gallery - Create a responsive image gallery by using the free Unsplash API"

and the second thing is to link your PR with the issue by adding the below line in the PR description.

Example:

Fixes #910

**Comment by siddhantsiddh15** [comment]
Greetings,

I have updated the pull request name and linked the issue with the issue number 
image

Thanks

**Comment by Sachin-chaurasiya** [comment]
@siddhantsiddh15 , Please format and lint the code by following this guide.
https://github.com/reactplay/react-play#format-the-code

**Comment by siddhantsiddh15** [comment]
Greetings

Have formatted the code as per the guidelines as mentioned here [https://github.com/reactplay/react-play#format-the-code](https://github.com/reactplay/react-play#format-the-code) and then pushed the code in the branch. 

Thanks

File: src/plays/image-gallery-using-unsplash-api/styles.css
Comment (atapas) [inline src/plays/image-gallery-using-unsplash-api/styles.css:2]: Please do not override the styles this way. Add a unqie class name in front of all the styles in all the CSS files so that the override changes are less.

You can add an additional style name here and use that to prefix all styles in all CSS files.

// src/plays/image-gallery-using-unsplash-api/ImageGalleryUsingUnsplashApi.js
```html
<div className="play-details-body">
```

**Review by atapas — CHANGES_REQUESTED** [review]
@siddhantsiddh15 Review comments

also please record your demo on stackstream and add the link by editing your play. Thanks

File: src/plays/image-gallery-using-unsplash-api/styles.css
Comment (siddhantsiddh15) [inline src/plays/image-gallery-using-unsplash-api/styles.css:2]: Have made changes here. Earlier it was *{ padding: 0;
  margin: 0;
  font-family: 'IBM Plex Sans', sans-serif;
  background-color: whitesmoke;
} .

Now it is under a className body.

**Comment by atapas** [comment]
@siddhantsiddh15 Kindly resolve the merge conflict

File: src/plays/image-gallery-using-unsplash-api/ImageGalleryUsingUnsplashApi.js
Comment (atapas) [inline src/plays/image-gallery-using-unsplash-api/ImageGalleryUsingUnsplashApi.js:99]: I would certainly change the class name from body to something more unique :) How about `unsplash-ig`?

File: src/plays/image-gallery-using-unsplash-api/styles.css
Comment (atapas) [reply src/plays/image-gallery-using-unsplash-api/styles.css:2]: The class name should prefix all the styles in the CSS file.

**Review by atapas — CHANGES_REQUESTED** [review]

**Comment by siddhantsiddh15** [comment]
image
These errors are not in my edited files, what can I do to run the react play?

**Comment by atapas** [comment]
@siddhantsiddh15 Catch me up on Discord today to close it.

File: src/plays/image-gallery-using-unsplash-api/styles.css
Comment (atapas) [inline src/plays/image-gallery-using-unsplash-api/styles.css:25]: No tag selector override plz.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/plays/image-gallery-using-unsplash-api/styles.css
Comment (siddhantsiddh15) [reply src/plays/image-gallery-using-unsplash-api/styles.css]: image
Hello Tapas, 

I have updated the styles.css folder and the code is now working fine without any errors.

Regards

**Comment by siddhantsiddh15** [comment]
[Link](https://youtu.be/EWhlpd5TW74) to the video as I was facing repeated issue in creating an account on Stack Stream.

**Comment by siddhantsiddh15** [comment]
Thankyou for merging my branch into main. Can I close this pull request now?

**Comment by atapas** [comment]
> Inspect

I have kicked off a build. Please check if it is successful and test if things are fine.

**Comment by atapas** [comment]
@siddhantsiddh15 here is the preview build.. I see the changes are breaking styles. Please take a look

https://react-play-git-fork-siddhantsiddh15-unsplash-8f3010-reactplayio.vercel.app/

**Comment by siddhantsiddh15** [comment]
Have recorded the video [here](https://youtu.be/oL8l_elY7qE) .  The website is responsive and is working correctly.

**Comment by atapas** [comment]
@siddhantsiddh15 almost there..

please add a cover image

image

Also edit your play from localhost and add the stream recording.

Ping where when you done, will merge it.

**Comment by atapas** [comment]
@siddhantsiddh15 let us know when done

**Comment by siddhantsiddh15** [comment]
Have added the cover image.
image

**Comment by atapas** [comment]
> Have added the cover image. image

Thanks!

- The cover image should be in KBs, please reduce the size.
- Also confirm that your demo recording has been added to the play by editing it.

**Comment by siddhantsiddh15** [comment]
Greetings,

I have not added the demo recording in the Play. I have updated the size of the cover.png to 13 kb. 

Regards

**Comment by atapas** [comment]
> Greetings,
> 
> I have not added the demo recording in the Play. I have updated the size of the cover.png to 135 kb.
> 
> Regards

Can you please add the demo recording too.. then its al done.

**Comment by siddhantsiddh15** [comment]
Cannot do it. Having difficulty. Can we skip the recording portion?

**Comment by atapas** [comment]
> Cannot do it. Having difficulty. Can we skip the recording portion?

Ok, no worries, no pressure. It's still valuable to get your work in.

I am just curious about what kind of issues you are facing. If you can post about it in our Discord, I may try the resolution.. In fact, you can add the recording after merge too...(before 5th March)

**Comment by siddhantsiddh15** [comment]
I am having slow internet connection due to the place I have travelled to,  recording is a big file to upload.

**Comment by koustov** [comment]
Hey @siddhantsiddh15 , this play looks cool. I will be waiting for the video link to updated before merging it to production branch

**Comment by siddhantsiddh15** [comment]
Okay, will update it by tonight.

**Comment by siddhantsiddh15** [comment]
> Hey @siddhantsiddh15 , this play looks cool. I will be waiting for the video link to updated before merging it to production branch

[Here](https://youtu.be/oNyJnWiE9RE) is the updated video of the play. I have checked the responsiveness and working of my play. The delay from my side was unwanted, I have uploaded the video as soon as I got good internet connection.

**Comment by koustov** [comment]
> > Hey @siddhantsiddh15 , this play looks cool. I will be waiting for the video link to updated before merging it to production branch
> 
> I have uploaded [here](https://youtu.be/oNyJnWiE9RE) the updated video of the play. I have checked the responsiveness and working of my play. The delay from my side was unwanted, I have uploaded the video as soon as I got good internet connection.

Well the video should be on https://stack-stream.com/ 
image

I can get this play merged if everything is ok however please record a stackstream video before EOD

**Comment by koustov** [comment]
@atapas  you need to unblock in order to merge this PR
@siddhantsiddh15  please record demo on stackstream

**Review by atapas — APPROVED** [review]

**Pull request merged**
